### PR TITLE
PR to resolve tasks within #1039

### PR DESCRIPTION
**Pull request by JackScanlon**
See issue for details:

- #1039

### CODE REVIEW

**Comment by elmessary** [comment]
@JackScanlon 
I think this issue needs to be handled differently.
We can have  a meeting about it

File: CodeListLibrary_project/clinicalcode/permissions.py [hunk 1]
```diff
@@ -22,7 +35,67 @@ class Permissions:
...
+    from .models.Phenotype import Phenotype
+    from .models.PhenotypeWorkingset import PhenotypeWorkingset
+    from .models.PublishedConcept import PublishedConcept
+    from .models.PublishedPhenotype import PublishedPhenotype
+    from .models.PublishedWorkingset import PublishedWorkingset
+
+    user = request.user
+    entity = set_cls.objects.get(pk=pk)
+    
+    if user.is_superuser or entity.owner == user or is_member(user, "Moderators"):
```
Comment (elmessary) [inline CodeListLibrary_project/clinicalcode/permissions.py:83]: this needs to be through permission as we have other conditions for access

Comment (JackScanlon) [reply CodeListLibrary_project/clinicalcode/permissions.py]: This is there just to determine whether to traverse through the historical records, the actual permision checks occur within the phenotype/concept/phenotypeworkingset views through the call to the method ```validate_access_to_view``` on lines 373, 188 and 579 respectively.
 
I guess there is a possibility that some cases may not be caught that may require traversal, so I added ```allowed_to_view``` method calls into this PR though as you requested

**Comment by elmessary** [comment]
implemented

**Pull request closed without merging**